src/include/daos_srv/dtx_srv.h

```diff
@@ -74,12 +83,20 @@ struct dtx_handle {
 	void				*dth_ent;
 	/** The flags, see dtx_entry_flags. */
 	uint16_t			 dth_flags;
-
 	/** Modification sequence in the distributed transaction. */
 	uint16_t			 dth_op_seq;
-	/** The intent of related modification. */
-	uint32_t			 dth_intent;
-	/* The hash of the dkey to be modified if applicable */
+
+	/** The count of EC objects that are modified by this DTX. */
+	uint16_t			 dth_ec_oid_cnt;
+	/** The total slots in the dth_ec_oid_array. */
+	uint16_t			 dth_ec_oid_cap;
+	/**
+	 * If more than one EC objects are modified, the IDs are reocrded here,
+	 * mainly used for EC aggregation.
+	 */
+	daos_unit_oid_t			*dth_ec_oid_array;
```

Comment:
Minor: Is this dedicated for EC? If they are for general purpose, it's better to not using 'ec' in var names. (DTX should have no idea of EC, right?)

Reply:
It is only used for EC. I agree with you that the VOS has no idea about "EC", but maintain the EC objects ID array is not free, so the upper layer "DTX" will not generate such array for non-EC objects.